src/ripple/protocol/Serializer.h

```diff
@@ -298,7 +297,7 @@ class Serializer
     }
 
     // low-level VL length encode/decode functions
-    static Blob encodeVL (int length);
+    static int encodeVL (int length, std::array<std::uint8_t, 4>& result);
```

Comment:
Hi Scott, this looks like a great approach to not using a vector, especially since the (slightly odd) signature doesn't seem to go outside of `Serializer`.  Another possible signature would be to return a `std::pair<int, std::array<...>>`, but I'm not convinced that's any more intuitive.  The return value optimization would make the two signatures approximately the same cost in execution time.

It looks like you were able to make this change without changing any code outside of `Serializer.h` and `Serializer.cpp`.  Could this method be made `private`?  If it could, that suggests some of the other methods could be `private` as well.  You might consider expanding the scope of the pull request by doing an audit of `Serializer` and making as many methods `private` as possible.  Just a suggestion.